pkg/cmd/codespace/code.go

```diff
@@ -5,37 +5,50 @@ import (
 	"fmt"
 	"net/url"
 
+	"github.com/cli/cli/v2/utils"
 	"github.com/spf13/cobra"
 )
 
 func newCodeCmd(app *App) *cobra.Command {
 	var (
 		codespace   string
 		useInsiders bool
+		useWeb      bool
 	)
 
 	codeCmd := &cobra.Command{
 		Use:   "code",
 		Short: "Open a codespace in Visual Studio Code",
 		Args:  noArgsConstraint,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return app.VSCode(cmd.Context(), codespace, useInsiders)
+			return app.VSCode(cmd.Context(), codespace, useInsiders, useWeb)
```

Comment:
I just realized that we are not respecting the `useInsiders` flag in case someone provides both that + the new `web` option. TIL that [VS Code for Web has an insiders build](https://code.visualstudio.com/docs/editor/vscode-web#_relationship-to-vs-code-desktop) at https://insiders.vscode.dev/

Although it doesn't look like that works for Codespaces when I tried it out. We should probably add a validation that checks only one of those flags is present otherwise provide an error to the user right above this.

Reply:
There is an optional query param `vscodeChannel` that you can use to select the between `insiders` and `stable` - e.g. `https://<name>.github.dev/?vscodeChannel=insiders` - in order to handle the `--insiders` flag. I think you'd be able to  do something like `codespace.WebUrl + "?vscodeChannel=insiders"` below.